### PR TITLE
@UiElement annotation feature

### PR DESCRIPTION
**Pull request by peanutbother**
Added load method to bind models to xml files in UiXmlLoader.java
Added Annotation to mark class fields as @UiElement similar to @Field

### CODE REVIEW

File: ui/src/main/java/org/mini2Dx/ui/xml/UiXmlLoader.java [hunk 1]
```diff
@@ -116,6 +119,39 @@ public void addTagHandler(String tagName, UiElementFactory factory, UiElementPop
...
+        if (container != null) {
+            return populateModel(container, model);
+        } else {
+            throw new MdxException("Failed to populate Ui file" + filename, new Exception("container is null"));
+        }
+    }
+
+    private <T extends UiElement, M> M populateModel(T container, Class<M> model) {
+        try {
+            M newInstance = model.getConstructor().newInstance();
```
Comment (tomcashman) [inline ui/src/main/java/org/mini2Dx/ui/xml/UiXmlLoader.java:133]: Use `Mdx.reflect.newInstance` here

File: ui/src/main/java/org/mini2Dx/ui/xml/UiXmlLoader.java [hunk 2]
```diff
@@ -116,6 +119,39 @@ public void addTagHandler(String tagName, UiElementFactory factory, UiElementPop
...
+            for (Field field : model.getDeclaredFields()) {
```
Comment (tomcashman) [inline ui/src/main/java/org/mini2Dx/ui/xml/UiXmlLoader.java:134]: Use `Mdx.reflect.getDeclaredFields` here

File: ui/src/main/java/org/mini2Dx/ui/xml/UiXmlLoader.java [hunk 3]
```diff
@@ -116,6 +119,39 @@ public void addTagHandler(String tagName, UiElementFactory factory, UiElementPop
...
+                if (field.isAnnotationPresent(org.mini2Dx.ui.annotation.UiElement.class)) {
```
Comment (tomcashman) [inline ui/src/main/java/org/mini2Dx/ui/xml/UiXmlLoader.java:135]: `Mdx.reflect.isAnnotationPresent`

File: ui/src/main/java/org/mini2Dx/ui/xml/UiXmlLoader.java [hunk 4]
```diff
@@ -116,6 +119,39 @@ public void addTagHandler(String tagName, UiElementFactory factory, UiElementPop
...
+                    String fieldName = field.getName();
+                    String annotationId = field.getAnnotation(org.mini2Dx.ui.annotation.UiElement.class).id();
```
Comment (tomcashman) [inline ui/src/main/java/org/mini2Dx/ui/xml/UiXmlLoader.java:137]: `Mdx.reflect.getAnnotation`

File: ui/src/main/java/org/mini2Dx/ui/xml/UiXmlLoader.java (hunk 1 above)
Comment (peanutbother) [reply ui/src/main/java/org/mini2Dx/ui/xml/UiXmlLoader.java:133]: I will update this, but why 
```suggestion
            M newInstance = (M) Mdx.reflect.newInstance(model);
```

this introduces an unchecked cast and uses deprecated `Class.getInstance()`
(the latter can be updated, but the former is just a new possible breaking point)

Comment (tomcashman) [reply ui/src/main/java/org/mini2Dx/ui/xml/UiXmlLoader.java:133]: We can update the Class.getInstance() but the Mdx.reflect API is to allow for cross-platform bindings for games consoles where reflection isn't supported and needs to be computed ahead-of-time.

**Pull request merged**